backend/models/sequence.js

```diff
@@ -116,24 +224,36 @@ Sequence.getSequenceState = async (account, model, stateId) => {
 };
 
 Sequence.getList = async (account, model, branch, revision, cleanResponse = false) => {
-	const history = await History.getHistory(account, model, branch, revision);
+	let submodelBranch;
+	let sequencesQuery = {};
+
+	if (branch || revision) {
+		const history = await History.getHistory(account, model, branch, revision);
+
+		if (!history) {
+			throw responseCodes.INVALID_TAG_NAME;
+		}
 
-	if(!history) {
-		throw responseCodes.INVALID_TAG_NAME;
+		submodelBranch = "master";
+		sequencesQuery = {"$or":[{"rev_id": history._id}, {"rev_id": {"$exists": false}}]};
 	}
 
-	const refNodes = await getRefNodes(account, model, branch, revision, {project:1});
+	const refNodesBranch = revision ? undefined : "master";
+	const refNodes = await getRefNodes(account, model, refNodesBranch, revision, {project:1});
 	const submodels = refNodes.map(r => r.project);
 
-	const submodelSequencesPromises = Promise.all(submodels.map((submodel) => Sequence.getList(account, submodel, "master", null, cleanResponse)));
+	const submodelSequencesPromises = Promise.all(submodels.map((submodel) => Sequence.getList(account, submodel, submodelBranch, undefined, cleanResponse)));
+
+	const sequences = await db.find(account, sequenceCol(model), sequencesQuery);
 
-	const sequences = await db.find(account, sequenceCol(model), {"rev_id": history._id});
 	sequences.forEach((sequence) => {
-		sequence.teamspace = account;
-		sequence.model = model;
+		sequence.minDate = new Date((sequence.frames[0] || {}).dateTime).getTime();
```

Comment:
as discussed, we should just write it into the db if we need it.

The whole purpose of ditching the frames was to reduce the data, so if we're fetching it to the backend it kind of defeats the point

Reply:
Added it on the frontend side but just putting here for consistency too... can we rename to startDate and endDate?